### ISSUE TITLE
Safe html filter - adding valid tags not working as expected, tags still filtered

### ISSUE THREAD

**Issue by fredvd** [plone/Products.CMFPlone#2693]
Summary: Plone 5.1.4 adding additional tags to the 'valid tags' field in the HTML Filter control panel prevents the tags being filtered on save when they are added to a Page, but the output/view still has the tags filtered. 

### Use Case:

Allowing a content editor to copy/paste a simple html form with 'form' and 'input' tags

### What I expect to happen:

When I add 'form' and 'input' to the valid tags that I can use these in a Page. 

### What actually happened:

After adding the two tags and re-edting the page, the form is visible in the Tinymce editor (edit screen) and are preserved inbetween saves. But the page view still has the tags filters. 

Only when I disable the whole HTML Filter (and re-edit) the form/input tags appear on view.

### What version of Plone/ Addons I am using:

5.1.4

**Comment by sureshvv** [comment]
I would like to help.

Can you provide a step-by-step outline to reproduce the problem?

I created a document with the <form> and <input> tags and they are displaying for me (as text).

**Comment by fredvd** [comment]
@sureshvv  Glad you wnat to help with this

>I created a document with the <form> and <input> tags and they are displaying for me (as text).

If the fields are displaying as text then there is something wrong, they should be displaying as the form elements.  Can you clarify what you mean with (as text)?

**Comment by fredvd** [comment]
@sureshvv to demonstrate, here are some screenshots from Plone 5.1.6.dev  (buildout coredev plone 5.1 branch)

Adding form and input tags to the safe_html filter:

<img width="339" alt="filter_settings" src="https://user-images.githubusercontent.com/394784/52709470-19210780-2f8d-11e9-9aa5-12849da9002e.png">

Adding a simple form to a page:

<img width="466" alt="form1" src="https://user-images.githubusercontent.com/394784/52709483-22aa6f80-2f8d-11e9-952f-df55aac2a4cf.png">

Viewing the page (after save):

<img width="446" alt="view" src="https://user-images.githubusercontent.com/394784/52709493-2a6a1400-2f8d-11e9-9859-048f4bae05dc.png">

If you edit the page again, you see the fields again. So TinyMCE isn't filtering the fields, but the .output of the RichText object is filtered.....

**Comment by mauritsvanrees** [comment]
I see what the problem is. In the safe_html transform we [use the `lxml.html.clean.Cleaner` class](https://github.com/plone/Products.PortalTransforms/blob/3.1.5/Products/PortalTransforms/transforms/safe_html.py#L2436). This class by default has an attribute `forms = True`. When True, this removes form tags, even when those tags are in the `valid_tags`:

```
if self.forms:
    remove_tags.add('form’)
    kill_tags.update(('button', 'input', 'select', 'textarea'))
```

See the [lxml code](https://github.com/lxml/lxml/blob/lxml-4.2.1/src/lxml/html/clean.py#L360-L362).

(Note: `remove_tags` removes a tag but keeps the contents, `kill_tags` removes the tag including all contents.)

So when creating an instance of the `Cleaner` class, we should set `cleaner.forms = False` when `form` is in the `valid_tags`. Maybe the same for button, input, select and textarea.
Actually, `forms=False` seems good always: the form/button/input/select/textarea (and label and option) will be either removed or killed anyway when they are not in the `valid_tags`.

I will create a PR.

**Comment by mauritsvanrees** [comment]
See PR https://github.com/plone/Products.PortalTransforms/pull/39

**Comment by mauritsvanrees** [comment]
This is released in 3.1.6.

**Comment by fredvd** [comment]
@mauritsvanrees thanks! :-)
